optuna/trial.py

```diff
@@ -702,7 +702,7 @@ def _adjust_discrete_uniform_high(name, low, high, q):
 
     r = high - low
 
-    if math.fmod(r, q) != 0:
+    if (decimal.Decimal(str(r)) % decimal.Decimal(str(q))) != 0:
```

Comment:
```suggestion
    if (decimal.Decimal(str(r)) % decimal.Decimal(str(q))) != decimal.Decimal('0'):
```

Reply:
Thank you for your proposal. Please proceed.